### PR TITLE
Fix some documentation warnings for MediaBrowser.XbmcMetadata

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

**Comment by cvium** [comment]
`Savers/BaseNfoSaver.cs(387,33): error CS8314: An expression of type 'T' cannot be handled by a pattern of type 'Video' in C# 7.0. Please use language version 7.1 or greater. [/home/vsts/work/1/s/MediaBrowser.XbmcMetadata/MediaBrowser.XbmcMetadata.csproj]`

File: MediaBrowser.XbmcMetadata/Savers/BaseNfoSaver.cs [hunk 1]
```diff
@@ -25,76 +25,81 @@ namespace MediaBrowser.XbmcMetadata.Savers
...
-                    "userrating",
-
-                    "countrycode"
-
-        }.ToDictionary(i => i, StringComparer.OrdinalIgnoreCase);
+        public const string DateAddedFormat = "yyyy-MM-dd HH:mm:ss";
+
+        public const string YouTubeWatchUrl = "https://www.youtube.com/watch?v=";
+
+        private static readonly CultureInfo _usCulture = CultureInfo.ReadOnly(new CultureInfo("en-US"));
```
Comment (EraYaN) [inline MediaBrowser.XbmcMetadata/Savers/BaseNfoSaver.cs]: This should probably be replace by InvariantCulture, just like we did in other spots in the code.

File: MediaBrowser.XbmcMetadata/Savers/AlbumNfoSaver.cs [hunk 2]
```diff
@@ -15,26 +15,24 @@ namespace MediaBrowser.XbmcMetadata.Savers
...
         public override bool IsEnabledFor(BaseItem item, ItemUpdateType updateType)
-        {
-            if (!item.SupportsLocalMetadata)
-            {
-                return false;
-            }
-
-            return item is MusicAlbum && updateType >= MinimumUpdateType;
-        }
+            => !item.SupportsLocalMetadata && item is MusicAlbum && updateType >= MinimumUpdateType;
```
Comment (dkanada) [inline MediaBrowser.XbmcMetadata/Savers/AlbumNfoSaver.cs]: invert boolean?

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs [hunk 3]
```diff
@@ -42,6 +35,20 @@ public BaseNfoParser(ILogger logger, IConfigurationManager config, IProviderMana
...
+        /// </summary>
+        protected ILogger Logger { get; }
+
+        protected IProviderManager ProviderManager { get; }
+
+        protected virtual bool SupportsUrlAfterClosingXmlTag => false;
+
+        protected virtual string MovieDbParserSearchString => "themoviedb.org/movie/";
+
+
```
Comment (dkanada) [inline MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs]: double newline

File: MediaBrowser.XbmcMetadata/Savers/AlbumNfoSaver.cs (hunk 2 above)
Comment (dkanada) [inline MediaBrowser.XbmcMetadata/Savers/AlbumNfoSaver.cs]: invert boolean?

File: MediaBrowser.XbmcMetadata/Savers/ArtistNfoSaver.cs [hunk 4]
```diff
@@ -14,26 +14,24 @@ namespace MediaBrowser.XbmcMetadata.Savers
...
         public override bool IsEnabledFor(BaseItem item, ItemUpdateType updateType)
-        {
-            if (!item.SupportsLocalMetadata)
-            {
-                return false;
-            }
-
-            return item is MusicArtist && updateType >= MinimumUpdateType;
-        }
+            => !item.SupportsLocalMetadata && item is MusicArtist && updateType >= MinimumUpdateType;
```
Comment (dkanada) [inline MediaBrowser.XbmcMetadata/Savers/ArtistNfoSaver.cs]: same here

File: MediaBrowser.XbmcMetadata/Savers/BaseNfoSaver.cs [hunk 5]
```diff
@@ -25,76 +25,81 @@ namespace MediaBrowser.XbmcMetadata.Savers
...
+
+            "isuserfavorite",
+            "userrating",
+
+            "countrycode"
+        };
+
+        // filters control characters but allows only properly-formed surrogate sequences
+        private const string _invalidXMLCharsRegex = @"(?<![\uD800-\uDBFF])[\uDC00-\uDFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x9F\uFEFF\uFFFE\uFFFF]";
+
```
Comment (dkanada) [inline MediaBrowser.XbmcMetadata/Savers/BaseNfoSaver.cs]: double newline again, can we add a rule to warn for this?

File: MediaBrowser.XbmcMetadata/Savers/SeriesNfoSaver.cs [hunk 6]
```diff
@@ -13,26 +14,30 @@ namespace MediaBrowser.XbmcMetadata.Savers
...
         public override bool IsEnabledFor(BaseItem item, ItemUpdateType updateType)
-        {
-            if (!item.SupportsLocalMetadata)
-            {
-                return false;
-            }
-
-            return item is Series && updateType >= MinimumUpdateType;
-        }
+            => item.SupportsLocalMetadata && item is Series && updateType >= MinimumUpdateType;
```
Comment (dkanada) [inline MediaBrowser.XbmcMetadata/Savers/SeriesNfoSaver.cs]: I think this one is actually correct, but you forgot to invert the boolean for the others.

File: MediaBrowser.XbmcMetadata/Savers/BaseNfoSaver.cs (hunk 5 above)
Comment (Bond-009) [reply MediaBrowser.XbmcMetadata/Savers/BaseNfoSaver.cs]: I think there is one, it's just a bit hard to spot with 13000 warnings.

**Comment by anthonylavado** [comment]
There will be a minor conflict with some of the changes in #1702, where Entrypoint.cs and AlbumNFOSaver.cs are both edited, with some of the same fixes seen in this PR.

File: MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs [hunk 7]
```diff
@@ -14,43 +14,43 @@ namespace MediaBrowser.XbmcMetadata.Savers
...
         public override bool IsEnabledFor(BaseItem item, ItemUpdateType updateType)
-        {
-            if (!item.SupportsLocalMetadata)
-            {
-                return false;
-            }
-
-            return item is Episode && updateType >= MinimumUpdateType;
-        }
+            => !item.SupportsLocalMetadata && item is Episode && updateType >= MinimumUpdateType;
```
Comment (dkanada) [inline MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs]: still need to invert this one

Comment (anthonylavado) [reply MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs]: @Bond-009 Thoughts?

Comment (EraYaN) [reply MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs]: It seems to match the old code now

**Comment by joshuaboniface** [comment]
@dkanada Any further thoughts or is this good?

File: MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs (hunk 7 above)
Comment (dkanada) [reply MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs]: @joshuaboniface this was definitely not fixed, the opposite change is in multiple locations. Either this section is wrong or the other sections like [this](https://github.com/Bond-009/jellyfin/blob/3eca8b9c98e5ad0ad9cefd85f27439f05f5bd6b5/MediaBrowser.XbmcMetadata/Savers/AlbumNfoSaver.cs#L39) one are wrong.

Comment (Bond-009) [reply MediaBrowser.XbmcMetadata/Savers/EpisodeNfoSaver.cs]: @dkanada Sorry, fixed in #1775

**Pull request merged**